homeassistant/components/aemet/sensor.py

```diff
@@ -51,172 +83,262 @@
     ATTR_API_WIND_MAX_SPEED,
     ATTR_API_WIND_SPEED,
     ATTRIBUTION,
+    CONDITIONS_MAP,
     DOMAIN,
     ENTRY_NAME,
     ENTRY_WEATHER_COORDINATOR,
-    FORECAST_MODE_ATTR_API,
-    FORECAST_MODE_DAILY,
-    FORECAST_MODES,
-    FORECAST_MONITORED_CONDITIONS,
-    MONITORED_CONDITIONS,
 )
+from .entity import AemetEntity
 from .weather_update_coordinator import WeatherUpdateCoordinator
 
-FORECAST_SENSOR_TYPES: tuple[SensorEntityDescription, ...] = (
-    SensorEntityDescription(
-        key=ATTR_API_FORECAST_CONDITION,
-        name="Condition",
+
+@dataclass
+class AemetSensorEntityDescription(SensorEntityDescription):
+    """A class that describes AEMET OpenData sensor entities."""
+
+    keys: list[str] | None = None
+
```

Comment:
```suggestion
    value_fn: Callable[[Any],Any] = lambda: value: value

```

Please set the correct data and return type in the Callable

Reply:
Done in f064f9e010966a5c9832823cc324e04c7e348d2a, but I had to add `# pylint: disable=unnecessary-lambda` in order to prevent pylint from complaining:
`
************* Module homeassistant.components.aemet.sensor
homeassistant/components/aemet/sensor.py:110:17: W0108: Lambda may not be necessary (unnecessary-lambda)
homeassistant/components/aemet/sensor.py:117:17: W0108: Lambda may not be necessary (unnecessary-lambda)
homeassistant/components/aemet/sensor.py:177:17: W0108: Lambda may not be necessary (unnecessary-lambda)
homeassistant/components/aemet/sensor.py:185:17: W0108: Lambda may not be necessary (unnecessary-lambda)
homeassistant/components/aemet/sensor.py:231:17: W0108: Lambda may not be necessary (unnecessary-lambda)
homeassistant/components/aemet/sensor.py:292:17: W0108: Lambda may not be necessary (unnecessary-lambda)
homeassistant/components/aemet/sensor.py:332:17: W0108: Lambda may not be necessary (unnecessary-lambda)
`

Reply:
> Done in [f064f9e](https://github.com/home-assistant/core/commit/f064f9e010966a5c9832823cc324e04c7e348d2a), but I had to add `# pylint: disable=unnecessary-lambda` in order to prevent pylint from complaining: `************* Module homeassistant.components.aemet.sensor homeassistant/components/aemet/sensor.py:110:17: W0108: Lambda may not be necessary (unnecessary-lambda) homeassistant/components/aemet/sensor.py:117:17: W0108: Lambda may not be necessary (unnecessary-lambda) homeassistant/components/aemet/sensor.py:177:17: W0108: Lambda may not be necessary (unnecessary-lambda) homeassistant/components/aemet/sensor.py:185:17: W0108: Lambda may not be necessary (unnecessary-lambda) homeassistant/components/aemet/sensor.py:231:17: W0108: Lambda may not be necessary (unnecessary-lambda) homeassistant/components/aemet/sensor.py:292:17: W0108: Lambda may not be necessary (unnecessary-lambda) homeassistant/components/aemet/sensor.py:332:17: W0108: Lambda may not be necessary (unnecessary-lambda)`

Fixed in 8ff97f56c900ce021136fe38201186c0a149e400